test/extensions/filters/http/common/fuzz/uber_filter.cc

```diff
@@ -75,23 +73,90 @@ void UberFilterFuzzer::decode(Http::StreamDecoderFilter* filter, const test::fuz
       end_stream = true;
     }
     Buffer::OwnedImpl buffer(data_chunks[i]);
-    ENVOY_LOG_MISC(debug, "Decoding data (end_stream={}): {} ", end_stream, buffer.toString());
-    if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
+    const auto& dataStatus = sendData(filter, buffer, end_stream);
+    if (dataStatus != Http::FilterDataStatus::Continue) {
+      ENVOY_LOG_MISC(debug, "Finished with FilterDataStatus: {}", dataStatus);
       return;
     }
   }
 
   if (data.has_trailers()) {
-    ENVOY_LOG_MISC(debug, "Decoding trailers: {} ", data.trailers().DebugString());
-    auto trailers = Fuzz::fromHeaders<Http::TestRequestTrailerMapImpl>(data.trailers());
-    filter->decodeTrailers(trailers);
+    sendTrailers(filter, data);
+  }
+}
+
+template <>
+Http::FilterHeadersStatus UberFilterFuzzer::sendHeaders(Http::StreamDecoderFilter* filter,
+                                                        const test::fuzz::HttpData& data,
+                                                        bool end_stream) {
+  request_headers_ = Fuzz::fromHeaders<Http::TestRequestHeaderMapImpl>(data.headers());
```

Comment:
Could this be consolidated in the `include_headers` arg?

Reply:
It could, but the values won't be realistic (it won't set the :method to GET). Are you suggesting changing that?

Reply:
+1 fine as is to have a realistic default instead of a dummy one (that's mostly there to prevent nullptr derefs)